### PR TITLE
More complete Dispose

### PR DESCRIPTION
**Pull request by mstv**
## Proposed changes

- dispose dangling `_findAndReplaceForm`
- dispose `_cancellationTokenSequence` also if `components` are `null`
- add `_loadingControlAsync` and `_loadingControlSync` only once to `Controls`
- consider every field in `RevisionGridControl.Dispose`
- dispose dangling `_revisionFilter`

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/Editor/FileViewerInternal.cs [hunk 1]
```diff
@@ -36,6 +36,14 @@ public partial class FileViewerInternal : GitModuleControl, IFileViewer
 
         public FileViewerInternal()
         {
+            Disposed += (sender, e) =>
+            {
+                //// _diffHighlightService not disposable
+                //// _lineNumbersControl not disposable
+                //// _currentViewPositionCache not disposable
+                _findAndReplaceForm.Dispose();
+            };
```
Comment (sharwell) [inline GitUI/Editor/FileViewerInternal.cs]: 💡 This should be registered after the `Initialize` methods.

Comment (mstv) [reply GitUI/Editor/FileViewerInternal.cs]: Why? In order to be safe, the cleanup should be registered before something can go wrong.

Comment (sharwell) [reply GitUI/Editor/FileViewerInternal.cs]: If something goes wrong in the constructor (i.e. an exception is thrown), this event will not be called.

Comment (mstv) [reply GitUI/Editor/FileViewerInternal.cs]: If you like. Moved after the `Initialize` methods and rebased.

**Pull request merged**